datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1341,50 +1245,132 @@ fn from_substrait_null(null_type: &Type) -> Result<ScalarValue> {
     }
 }
 
-async fn make_datafusion_like(
-    case_insensitive: bool,
-    f: &ScalarFunction,
-    input_schema: &DFSchema,
-    extensions: &HashMap<u32, &String>,
-) -> Result<Arc<Expr>> {
-    let fn_name = if case_insensitive { "ILIKE" } else { "LIKE" };
-    if f.arguments.len() != 3 {
-        return not_impl_err!("Expect three arguments for `{fn_name}` expr");
+/// Build [`Expr`] from its name and required inputs.
+struct BuiltinExprBuilder {
+    expr_name: String,
+}
+
+impl BuiltinExprBuilder {
+    pub fn try_from_name(name: &str) -> Option<Self> {
+        match name {
```

Comment:
should we that be case sensitive?

Reply:
Looks like the existing code is also case sensitive. Not that we shouldn't fix it if it should be ignoring case, but I dont' think this PR makes it any better or worse

Reply:
Thanks for reminding me 👍 I added an item to #8149, I would take other implementations as a reference on how they handle case sensitivity.